kernel/scheduler.c

```diff
@@ -219,6 +222,8 @@ struct __attribute__((packed)) task_block {
 
     /* threads */
     struct task **threads;
+    struct task *joined_thread;
```

Comment:
It's not about the joinee, it's about the joiner. 

A creates B. A calls `pthread_join(B)`. A dies. B spawns a new thread C. The allocator reuses the same memory from the old (freed) 'A' ptr to allocate C. B->joiner now point to D, which is wrong.

If instead of a pointer for `joiner` field, we use the `tid` value of the joiner, this is much less likely to happen.